runhouse/resources/hardware/launcher_utils.py

```
@@ -356,6 +318,29 @@ def teardown(cls, cluster, verbose: bool = True):
             )
         cluster._cluster_status = ClusterStatus.TERMINATED
 
+    @classmethod
+    def load_creds(cls):
+        """Loads the SSH credentials required for the Den launcher, and for interacting with the cluster
+        once launched."""
+        default_ssh_key = rns_client.default_ssh_key
+        if not default_ssh_key:
+            raise ValueError(
+                "No default SSH key found in the local Runhouse config, "
+                "please set one by running `runhouse login`"
+            )
+        try:
+            # Note: we still need to load them down locally to use for certain cluster operations (ex: rsync)
+            secret = rh.Secret.from_name(default_ssh_key)
+            # Ensure this specific keypair is written down locally
+            secret.write(overwrite=True)
```

a bit hesitant on having overwrite be True, maybe throw an error instead that a secret exists in the location locally?

or if it's easy to wire up diff local vs saved file names, could generate a new unique file location locally (e.g. `~/.rh/ssh/cluster-name-ssh-key`). not sure if that'll lead to complications elsewhere tho (but also don't want to repeat saving the same creds under diff names)

---

hmm ya i see your point, the ore requiremnt here is that the key pair used in launch matches what's saved down locally in the sky path (~/.ssh/sky-key). waht aobut checking if the path exists locally, and only if it doesn't exist then writing those values down?

---

yea that sounds good, actually looking at `_write_to_file` for `SSHSecret1 we don't support `overwrite` for ssh anyways, and it currently throws a warning if there's a mismatch, we want to catch/raise the warning here or even make it an error

---

oh cool, so in that case having that method raise an error and letting that propagate back seems reasonable